src/context/StatsContext.jsx

```diff
@@ -2,111 +2,132 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import statsClient from 'panoptes-client/lib/stats-client';
 
-import { config } from '../config';
-
 export const StatsContext = React.createContext();
 
 export class StatsProvider extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      collectiveStatsByDay: [],
-      userStatsByDay: [],
-      userStatsByMonth: []
+      collectiveStatsByDay: new Map(),
+      userStatsByDay: new Map(),
+      userStatsByMonth: new Map()
     };
 
     this.fetchCollectiveStats = this.fetchCollectiveStats.bind(this);
   }
 
   componentDidMount() {
-    const { project, explorer } = this.props;
-    if (project && explorer) {
+    const { projects, explorer } = this.props;
+    if (projects && explorer) {
       this.fetchUserStats();
     }
   }
 
   componentDidUpdate(prevProps) {
-    const { project, explorer } = this.props;
-    if (prevProps.project !== project || prevProps.explorer !== explorer) {
+    const { projects, explorer } = this.props;
+    if (prevProps.projects !== projects || prevProps.explorer !== explorer) {
       this.fetchUserStats();
     }
   }
 
   fetchUserStats() {
-    const { explorer, project } = this.props;
-    if (project && explorer) {
-      statsClient
-        .query({
-          period: 'day',
-          projectID: config.projectId,
-          type: 'classification',
-          userID: explorer.id
-        })
-        .then(data =>
-          data.map(statObject => ({
-            label: statObject.key_as_string,
-            value: statObject.doc_count
-          }))
-        )
-        .then(statData => {
-          this.setState({ userStatsByDay: statData });
-        })
-        .catch(() => {
-          if (console) {
-            console.warn('Failed to fetch user daily stats.');
-          }
-        });
-      statsClient
-        .query({
-          period: 'month',
-          projectID: config.projectId,
-          type: 'classification',
-          userID: explorer.id
-        })
-        .then(data =>
-          data.map(statObject => ({
-            label: statObject.key_as_string,
-            value: statObject.doc_count
-          }))
-        )
-        .then(statData => {
-          this.setState({ userStatsByMonth: statData });
-        })
-        .catch(() => {
-          if (console) {
-            console.warn('Failed to fetch user monthly stats.');
-          }
-        });
-    } else {
-      this.setState({
-        collectiveStatsByDay: [],
-        userStatsByDay: [],
-        userStatsByMonth: []
+    const { explorer, projects } = this.props;
+    if (explorer && explorer.id && projects && projects.length) {
+      projects.forEach(project => {
+        statsClient
```

Comment:
I could be wrong, but I think the old stats service is being retired in favour of the new GraphQL service, so the queries here might change in future.

Reply:
Towards top of to dos is refactor for new GraphQL Stats service for few reasons.